mlflow/pyfunc/scoring_server/utils.py

```diff
@@ -0,0 +1,38 @@
+import logging
+from typing import Any, Dict, Union
+
+import mlflow
+from mlflow.exceptions import MlflowException
+from mlflow.pyfunc.scoring_server import _parse_json_data
+
+_logger = logging.getLogger(__name__)
+
+
+# NB: this function should always be kept in sync with the serving
+# process in scoring_server invocations.
+def _validate_serving_input(model_uri: str, serving_input: Union[str, Dict[str, Any]]):
+    """
+    Helper function to validate the model can be served and provided input is valid
+    prior to serving the model.
+
+    Args:
+        model_uri: URI of the model to be served.
+        serving_input: Input data to be validated. Should be a dictionary or a JSON string.
+
+    Returns:
+        The prediction result from the model.
+    """
+
+    try:
+        # sklearn model might not have python_function flavor if it
+        # doesn't define a predict function. In such case the model
+        # can not be served anyways
+        pyfunc_model = mlflow.pyfunc.load_model(model_uri)
+    except Exception as e:
+        raise MlflowException(f"Failed to load model for serving validation: {e}")
```

Comment:
do we need this try-catch?

Reply:
I see the benefit of alerting users to why a pyfunc load operation would raise when they're trying to log a model could be potentially confusing, if for no other reason than to reduce confusion. 
While it seems a bit of a stretch that a logged model can't be reloaded as a pyfunc (unless we shipped a regression), this seems like a positive notification decoration to a potentially confusing exception.

Reply:
I'm just not a big fan of error capturing like this. It hides the original stack traceback and doesn't add any useful information.

Reply:
@serena-ruan not a blocker, fine to keep it.

Reply:
Yes except for sklearn models (in the comment), I think all flavors can be loaded as pyfunc. I can remove this try catch as we have try catch in model logging anyways :)

Reply:
Ah, sorry @harupy I didn't notice the missing `raise MlflowException() from e` <- we should pass the original stack trace through when doing this :) 
Also fine to remove it entirely and let logging handle that in isolation!